packages/vscode-extension/src/debug/taskTerminal/devTunnelTaskTerminal.ts

```diff
@@ -133,51 +142,48 @@ export class DevTunnelTaskTerminal extends BaseTunnelTaskTerminal {
   }
 
   private async deleteExistingTunnel(): Promise<void> {
-    try {
-      if (!this.args?.output?.id) {
-        return;
-      }
-
-      const envsRes = await this.readPropertiesFromEnv(this.args.env);
-      if (envsRes.isErr()) {
-        return;
-      }
-      const id = envsRes.value[this.args.output.id];
-
-      const idArr = id?.split(".");
-      if (!idArr || idArr.length !== 2) {
-        return;
-      }
-
-      const tunnelInstance = await this.tunnelManagementClientImpl.getTunnel({
-        tunnelId: idArr[0],
-        clusterId: idArr[1],
-      });
+    const devTunnelStates = await this.devTunnelStateManager.listDevTunnelStates();
+    for (const devTunnelState of devTunnelStates) {
+      try {
+        if (devTunnelState.sessionId === vscode.env.sessionId) {
+          continue;
+        }
 
-      if (tunnelInstance?.tags?.includes(DevTunnelTag)) {
-        await this.tunnelManagementClientImpl.deleteTunnel(tunnelInstance);
+        const tunnelInstance = await this.tunnelManagementClientImpl.getTunnel({
+          tunnelId: devTunnelState.tunnelId,
+          clusterId: devTunnelState.clusterId,
+        });
+        if (tunnelInstance?.tags?.includes(DevTunnelTag)) {
```

Comment:
Delete tunnel state when `tunnelInstance` is undefined?

Reply:
Updated. 
Note: if the user switch the m365 account, the `tunnelInstance` will be `undefined` and the tunnel will be delete by dev tunnel service after 30 days.